x-pack/plugins/security/public/management/users/edit_user/create_user_page.test.tsx

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { render, fireEvent, waitFor } from '@testing-library/react';
+import { createMemoryHistory } from 'history';
+import { coreMock } from '../../../../../../../src/core/public/mocks';
+import { securityMock } from '../../../mocks';
+import { Providers } from '../users_management_app';
+import { CreateUserPage } from './create_user_page';
+
+jest.mock('@elastic/eui/lib/services/accessibility/html_id_generator', () => ({
+  htmlIdGenerator: () => () => `id-${Math.random()}`,
+}));
+
+describe('CreateUserPage', () => {
+  it('creates user when submitting form and redirects back', async () => {
```

Comment:
Can we add a tests for:
1) The new warning you added when creating a user with a duplicate username

2) The validation rules that were moved out of the old validator and into `user_form`